api/base/permissions.py

```diff
@@ -173,4 +173,6 @@ def has_object_permission(self, request, view, obj):
         if request.method in permissions.SAFE_METHODS:
             return resource.is_public or resource.can_view(auth)
         else:
+            if isinstance(resource, Preprint):
+                return resource.can_edit(auth=auth)
```

Comment:
@Johnetordoff fyi, I decided to keep the code the same for nodes and registrations, only call `.can_edit()` for preprints to avoid accidentally break others. Since the `can_edit()` is much more complex than `return resource.has_permission(auth.user, osf_permissions.WRITE)`.